### PR TITLE
Features/authentication

### PR DESCRIPTION
**Pull request by gearprn**

### CODE REVIEW

File: frontend/src/components/login/page/LoginPage.tsx [hunk 1]
```diff
@@ -14,6 +14,10 @@ const Login: FC = () => {
   const [bodyMessage, setBodyMessage] = useState('');
   const { isShowing, toggle } = useModal(false);
 
+  const handleModalCallBack = (stats: boolean) => {
+    if (!stats) toggle();
+  };
```
Comment (Icyscools) [inline frontend/src/components/login/page/LoginPage.tsx]: Change typo from `stats` to `status`

File: frontend/src/components/register/page/RegisterPage.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [displayName, setDisplayName] = useState('');
+  const [title, setTitle] = useState('');
+  const [bodyMessage, setBodyMessage] = useState('');
+  const { isShowing, toggle } = useModal(false);
+
+  const handleModalCallBack = (stats: boolean) => {
+    if (!stats) toggle();
+  };
```
Comment (Icyscools) [inline frontend/src/components/register/page/RegisterPage.tsx]: Change typo from `stats` to `status`

File: frontend/src/context/SessionContext.tsx [hunk 3]
```diff
@@ -11,12 +13,20 @@ interface IUser {
   role: string;
 }
 
+enum EnumCartStatus {
+  WAITING = 'WAITING',
+  CHECKOUT = 'CHECKOUT',
+}
+
```
Comment (Icyscools) [inline frontend/src/context/SessionContext.tsx]: Can you put in `types/cart` ?

File: frontend/src/context/SessionContext.tsx [hunk 4]
```diff
@@ -46,6 +57,24 @@ export const SessionProvider: FC = ({ children }) => {
...
+    async (username, password, displayName) => {
+      const result = await register({ variables: { username, password, displayName } });
+
+      if (result.data.register.token) {
+        setCookie('token', result.data.register.token, { maxAge: 86400 });
+        setUser(result?.data?.register?.user);
+
+        await createCart({
+          variables: { userId: result?.data?.register?.user?._id, status: EnumCartStatus.WAITING },
+        });
```
Comment (Icyscools) [inline frontend/src/context/SessionContext.tsx]: Can create cart after register in backend ?

**Review by Icyscools — CHANGES_REQUESTED** [review]

File: frontend/src/context/SessionContext.tsx (hunk 3 above)
Comment (gearprn) [reply frontend/src/context/SessionContext.tsx]: @Icyscools Totally agree.

File: frontend/src/context/SessionContext.tsx (hunk 4 above)
Comment (gearprn) [reply frontend/src/context/SessionContext.tsx]: @Icyscools Totally agree.

File: frontend/src/context/SessionContext.tsx [hunk 5]
```diff
@@ -25,11 +29,11 @@ export const SessionProvider: FC = ({ children }) => {
   const [, setCookie, removeCookie] = useCookies(['token']);
   const history = useHistory();
   const location = useLocation();
-
-  const [queryMe, { loading, data, client }] = useLazyQuery(ME_QUERY, {
+  const [login] = useMutation(LOGIN_MUTATION);
+  const [register] = useMutation(REGISTER_MUTATION);
+  const [queryMe, { data, client }] = useLazyQuery(ME_QUERY, {
```
Comment (Icyscools) [inline frontend/src/context/SessionContext.tsx]: Can you add type for `useMutation` and `useLazyQuery` on these lines ?
Ref: [https://www.apollographql.com/docs/react/development-testing/static-typing/#usemutation]( https://www.apollographql.com/docs/react/development-testing/static-typing/#usemutation)

**Review by Icyscools — APPROVED** [review]
Praise the lord @gearprn.

**Pull request merged**